### ISSUE TITLE
url alias not available in forum edit page

### ISSUE THREAD

**Issue by cptX** [backdrop-contrib/forum#31]
Hi, after having tried Drupal 9 I ended up trying Backdrop as in D9 more and more unsolved issues came to the surface.
One of the important issues I need to fix now in Backdrop is to set the correct paths for the forum pages.
In D9 in every forum edit page there is a URL alias field. Although, the default core forum module has a bug on this in D9, after patching I was able to use it successfully. In Backdrop though there is no such field at all. How can I set a manual url alias for a specific forum?
This is not to be confused with the aliases for the corresponding taxonomies. Taxonomies have url alias fields but if you go to that url you don't get the forum page but the taxonomy page.

**Comment by indigoxela** [comment]
Hi @cptX,

> I need to fix now in Backdrop is to set the correct paths for the forum pages.
> In D9 in every forum edit page there is a URL alias field.

Is your question about actual paths or about path aliases?

The _default_ alias patterns can get configured on admin/config/urls/path/patterns - search for "Forum topic".
Additionally, the URL alias pattern for the content type can also get set on the content type form (admin/structure/types/manage/forum) in the section "Default URL alias pattern" (in the vertical tabs below description).

And the aliases can get overridden per Forum topic (depending on permissions). Again in the vertical tabs, this time section "URL settings".

_All_ aliases are available on this admin form: admin/config/urls/path

There you can also add new ones (Add URL alias).

Does this answer your question?

**Comment by cptX** [comment]
Hi, you are talking about topic alias. I'm talking about forum page alias and not to be confused with the corresponding taxonomy alias.
At the moment backdrop allows this only from the alias list which is not so user friendly. The user sees something like "taxonomy/term/10" which doesn't say which forum page it is.
Please consider including an alias field in every forum page.

**Comment by indigoxela** [comment]
> I don't want to use a pattern for the forum pages. I will probably use a pattern for the topics but not for the forum pages. So normally in D9 every forum edit page has an alias field where the user can set manually the alias.

With "forum pages" you mean the taxonomy term, right? Like "General discussion" for instance...

You can also set the alias per forum. It's the same form structure as with nodes.

Go to admin/structure/taxonomy/forums to get an overview list of all your forums, edit them individually as needed.
To change the alias, edit, scroll down to the vertical tabs in such an edit form - there's again the section "URL settings" - just like with nodes.

Their alias pattern can get edited on admin/config/urls/path/patterns - scroll down a bit, search for "URL alias pattern for all Forums URLs".

**Comment by cptX** [comment]
You are mistaken. The setting of the url path of the taxonomy is not the same as with the forum page. The forum page shows the forum table, the taxonomy page shows a raw list.

**Comment by cptX** [comment]
Issue reopened here https://github.com/backdrop/backdrop-issues/issues/4948

**Comment by cptX** [comment]
Reopening the issue here as it looks like it has nothing to do with the core.
Please see the following image from D9. The parts in the red box are completely missing from Backdrop forum edit pages now
![image](https://user-images.githubusercontent.com/9040042/108226021-2d340c80-7145-11eb-9054-863379576332.png)

**Comment by cptX** [comment]
Any progress or intention to work on this? Theoretically shouldn't be too hard to add this field.

**Comment by indigoxela** [comment]
> Theoretically shouldn't be too hard to add this field.

Have you ever thought about providing a pull request? :smile: (p.s. I'm not maintainer for this module).

How about this alternate module? https://github.com/backdrop-contrib/forum_ng

**Comment by cptX** [comment]
I knew you would answer like this. Yes I have considered getting deeper in the code but 1. I'm not considering myself a programmer and 2. I have no clue how this mechanism would work. I suppose I would check how other fields are created and I would check which hooks I could call in order to trigger the generation of the alias...
"Theoretically shouldn't be too hard to add this field." was meant for the maintainer who probably knows the forum module much better. If I had guidance I could check it too but at the moment my skills and available time to get these skills by myself do not allow this.

**Comment by cptX** [comment]
Visited forum_ng but didn't understand what is different than the core forum. Any clues?

**Comment by indigoxela** [comment]
> Visited forum_ng but didn't understand what is different than the core forum.

The new forum_ng is views-based. The old (not in core anymore) forum is very similar to the old D7 forum.
So the new one shouldn't contain the oddities you're struggling with. I didn't use either module yet, but if I set up a new Backdrop site with a forum, I'd definitely give forum_ng a try.

> I knew you would answer like this. ...

Hehe, sure you did. :smile: It's always worth a try to ask for a PR, no matter if it's successful or not.

**Comment by cptX** [comment]
Sounds really promising. Waiting an answer from the maintainers to understand if it is mature for production. Hope it is because the standard forum module is not adequate! I'll give it a try!
Anyway, this doesn't mean the issue described here shouldn't be solved. It is solved in D8/D9 so I think for the completeness of the system would be good to have it in BD too
